src/Microsoft.CodeAnalysis.NetAnalyzers/src/Microsoft.CodeAnalysis.NetAnalyzers/Microsoft.CodeQuality.Analyzers/Maintainability/UseCrossPlatformIntrinsicsAnalyzer.cs

```diff
@@ -140,169 +204,357 @@ private void OnCompilationStart(CompilationStartAnalysisContext context)
+            {
+                // Note: AdvSimd.Arm64.Abs for integer types returns unsigned types (e.g., Vector128<long> → Vector128<ulong>),
+                // so we only register AbsScalar for compatible types
+                AddUnaryMethods(methodSymbols, "AbsScalar", armAdvSimdArm64TypeSymbolForMethods, RuleKind.Abs, [SpecialType.System_Double]);
+                AddBinaryMethods(methodSymbols, "AddSaturateScalar", armAdvSimdArm64TypeSymbolForMethods, RuleKind.AddSaturate);
+                // Note: Negate is already registered as op_UnaryNegation above, so we don't register it here
+                // TODO: Add scalar variants of Extract/Insert, FusedMultiply*, Round*
+            }
+
+            if (compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeIntrinsicsWasmPackedSimd, out var wasmPackedSimdTypeSymbolForMethods))
+            {
+                AddUnaryMethods(methodSymbols, "Abs", wasmPackedSimdTypeSymbolForMethods, RuleKind.Abs);
+                AddUnaryMethods(methodSymbols, "Ceiling", wasmPackedSimdTypeSymbolForMethods, RuleKind.Ceiling);
+                AddBinaryMethods(methodSymbols, "CompareEqual", wasmPackedSimdTypeSymbolForMethods, RuleKind.Equals);
+                AddBinaryMethods(methodSymbols, "CompareGreaterThan", wasmPackedSimdTypeSymbolForMethods, RuleKind.GreaterThan);
+                AddBinaryMethods(methodSymbols, "CompareGreaterThanOrEqual", wasmPackedSimdTypeSymbolForMethods, RuleKind.GreaterThanOrEqual);
+                AddBinaryMethods(methodSymbols, "CompareLessThan", wasmPackedSimdTypeSymbolForMethods, RuleKind.LessThan);
+                AddBinaryMethods(methodSymbols, "CompareLessThanOrEqual", wasmPackedSimdTypeSymbolForMethods, RuleKind.LessThanOrEqual);
+                AddUnaryMethods(methodSymbols, "Floor", wasmPackedSimdTypeSymbolForMethods, RuleKind.Floor);
+                AddBinaryMethods(methodSymbols, "Max", wasmPackedSimdTypeSymbolForMethods, RuleKind.Max);
+                AddBinaryMethods(methodSymbols, "Min", wasmPackedSimdTypeSymbolForMethods, RuleKind.Min);
+                // Note: Negate is already registered as op_UnaryNegation above, so we don't register it here
+                AddUnaryMethods(methodSymbols, "Sqrt", wasmPackedSimdTypeSymbolForMethods, RuleKind.Sqrt);
+                AddUnaryMethods(methodSymbols, "Truncate", wasmPackedSimdTypeSymbolForMethods, RuleKind.Truncate);
+            }
+
+            if (compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeIntrinsicsX86Avx, out var x86AvxTypeSymbolForMethods))
+            {
+                AddBinaryMethods(methodSymbols, "AddSaturate", x86AvxTypeSymbolForMethods, RuleKind.AddSaturate);
+                AddBinaryMethods(methodSymbols, "AndNot", x86AvxTypeSymbolForMethods, RuleKind.AndNot_Swapped);
+                AddTernaryMethods(methodSymbols, "BlendVariable", x86AvxTypeSymbolForMethods, RuleKind.ConditionalSelect);
```

Comment:
`BlendVariable` is not valid to replace with `ConditionalSelect` due to a difference in behavior.

`BlendVariable` only considers the most significant bit of the mask when determining whether to select the entire element from left vs right. Where-as `ConditionalSelect` is bitwise and does this determination for each bit individually.

There are some cases of `mask` where you know this transform is safe, such as if it is a comparison known to produce `AllBitsSet` or `Zero` on a per-element basis. But it cannot be done for all instances.